### ISSUE TITLE
Performance issues on parallel chart 

### ISSUE THREAD

**Issue by mgalloue** [novus/nvd3#1334]
Hi @liquidpele, I'm still working on parallel chart side and I'm hitting performance issues. For example, on parallelCoordinate file, this part cost a lot (especially with many dimensions). 
 `//Restore axis brush
            dimensions.select('.nv-brushBackground')
                .each(function (d) {
                        d3.select(this).call(y[d.key].brush);
                })
                .selectAll('rect')
                .attr('x', -8)
                .attr('width', 16); `

Each time the dataSet is updated chart.update is called and this seems to be a hot point. This loop is needed in order to keep all brushes on axis.

Do you have any idea on what I can do to make this part less expensive?

Thanks a lot for your help.

**Comment by liquidpele** [comment]
Not really, that chart was in a deprecated folder before I started working with nvd3... I fixed a few bugs on it and made an example to use it since I thought it was still a useful chart, but I haven't really looked at the overall code quality there to be honest.   

If you have a lot of dimensions (how many are we talking here??) then I'd say that the d3.select and selectAll are probably expensive.   For instance, if the rect's created have static x and width, then there is no need to redo that every time, we can add them when they are created only.
